File: 12Mastering_Swift4/13Using Mix and Match/ObjectiveCExample/ObjectiveCExample/Messages.m
#import "Messages.h"

@implementation Messages

NSMutableArray *theMessages;
-(id)init {
    if ( self = [super init] ) {
        theMessages = [NSMutableArray new];
        [theMessages addObject:@"You should learn from your mistakes"];
        [theMessages addObject:@"It is in the now that we must live"];
        [theMessages addObject:@"The greatest risk is not taking one"];
        [theMessages addObject:@"You will be a Swift programmer"];
    }
    return self;
}

-(NSString *)getMessage {
    int num = arc4random() % theMessages.count;
    return theMessages[num];
}

@end

File: 12Mastering_Swift4/13Using Mix and Match/SwiftExample/SwiftExample/Messages.swift
import Foundation

class Messages: NSObject {
    
    private var messages = [String]()
    
    override init() {
        messages.append("You should learn from your mistakes")
        messages.append("It is in the now that we must live")
        messages.append("The greatest risk is not taking one")
        messages.append("You will be a Swift programmer")
    }
    
    @objc func getMessage() -> String {
        let num = Int(arc4random()) % messages.count
        return messages[num]
    }
}

